### PR TITLE
Pass data to view as second param

### PR DESCRIPTION
**Pull request by howdu**
Consistent with how Laravel's view method works and needing to know about `viewData()`.

ViewField->make()->view('filament.forms.components.range-slider', ['custom_data' => CustomData::get()])

### CODE REVIEW

File: packages/support/src/Components/ViewComponent.php [hunk 1]
```diff
@@ -32,10 +32,14 @@ abstract class ViewComponent extends Component implements Htmlable
 
     protected static array $methodCache = [];
 
-    public function view(string $view): static
+    public function view(string $view, array $viewData = []): static
     {
         $this->view = $view;
 
+        if ($viewData) {
```
Comment (zepfietje) [inline packages/support/src/Components/ViewComponent.php:39]: Isn't this check always going to be true if the default value is `[]`?

Comment (howdu) [reply packages/support/src/Components/ViewComponent.php:39]: Empty array is false in PHP

Comment (zepfietje) [reply packages/support/src/Components/ViewComponent.php:39]: Oh true, lol. Dan fixed this already after merging. 👍

Comment (zepfietje) [reply packages/support/src/Components/ViewComponent.php:39]: Would be an issue if you call `viewData` before `view` though, @danharrin.

**Pull request merged**